sdk/python/smithy_python/components/component.py

```diff
@@ -0,0 +1,132 @@
+from typing import Union, List, Optional
+from logging import Logger
+from smithy_python.remote_store.findings_service.v1 import findings_service_pb2
+from smithy_python.dbmanagers import RemoteDBManager, PostgresDBManager, SqliteDBManager
+from abc import ABC
+from smithy_python.enums.db_type_enum import DBTypeEnum
+
+
+class Component(ABC):
+    """
+    A component in the Smithy Python SDK that represents a component in the smithy framework.
+    """
+
+    def __init__(
+        self,
+        instance_id: Union[uuid.UUID, str],
+        db_type: DBTypeEnum,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        """
+        Initializes a new instance of the Component class.
+        **ATTENTION**: Currently only the `remote` database type is supported, all other types will raise a `NotImplementedError`.
+        :param instance_id: The UUID of the instance (aka the Workflow run to be analyzed)
+        :type instance_id: Union[uuid.UUID, str]
+        :param db_type: The database mode for the enricher, DBTypeEnum, can be either `SQLITE`, `POSTGRES` or `REMOTE`. (remote is gRPC)
+        :type db_type: DBTypeEnum
+        :param logger: Optional an instance of the Logger class for logging, if not provided, a default logger will be used.
+        :type logger: Optional[Logger]
+        :raises ValueError: If the instance_id is not a valid UUID.
+        :raises TypeError: If the instance_id is not a string or UUID object, or the logger is not an instance of Logger or None.
+        """
+
+        if not logger:
+            self.log = log
+        elif isinstance(logger, Logger):
+            self.log = logger
+        else:
+            raise TypeError("logger must be an instance of Logger or None.")
+
+        if instance_id is None or (
+            not isinstance(instance_id, str) and not isinstance(instance_id, uuid.UUID)
+        ):
+            raise TypeError("instance_id must be a string or UUID object.")
+
+        if isinstance(instance_id, uuid.UUID):
+            instance_id = str(instance_id)
+
+        if not self._validate_uuid(instance_id):
+            raise ValueError(f"Invalid UUID: {instance_id}")
+
+        self.instance_id = instance_id
+
+        if not db_type or not isinstance(db_type, DBTypeEnum):
+            raise TypeError("db_type must be a DBTypeEnum.")
+
+        match db_type:
+            case DBTypeEnum.SQLITE:
+                self.db_manager = SqliteDBManager()
+            case DBTypeEnum.POSTGRES:
+                self.db_manager = PostgresDBManager()
+            case DBTypeEnum.REMOTE:
+                self.db_manager = RemoteDBManager(instance_id=self.instance_id)
+            case _:
+                raise ValueError(
+                    "db_type must be one of DBTypeEnum.SQLITE, DBTypeEnum.POSTGRES, or DBTypeEnum.REMOTE."
+                )
+
+    def get_findings(
+        self, page_num: Optional[int] = None
+    ) -> Union[List[findings_service_pb2.Finding], List[any]]:
```

Comment:
why is the return value a list of `findings_service_pb2.Finding` or `any`? all these calls should use a replica of this type: https://github.com/smithy-security/smithy/blob/main/sdk/component/vulnerability-finding/finding.go#L1

basically the ocsf/v1.VulnerabilityFinding + the finding ID in a data class

Reply:
Ok, there is some duplication in our types that threw me off. We need to eventually clean this up but it's not a problem for now.

To answer your question, yes any DB driver is expected to eventually give us back an object of the protobuf type. How they do this, it's up to the drivers to figure out